framework/src/base/FEProblem.C

```diff
@@ -3499,6 +3499,9 @@ FEProblem::computeResidualType(const NumericVector<Number>& soln, NumericVector<
 
   execMultiApps(EXEC_LINEAR);
 
+  for (unsigned int tid = 0; tid < n_threads; tid++)
+    reinitScalars(tid);
```

Comment:
Yes, we do need that, see #7749. The reason is:

>Since user objects can couple scalar variables in, we should reinit the
values of all scalar variables before we execute the user object loop.

The same has to be done in `computeJacobian`